packages/botonic-react/tests/webchat/webchat.component.test.jsx

```diff
@@ -0,0 +1,163 @@
+import { render, screen } from '@testing-library/react'
+
+describe('TEST: webchat component', () => {
+  const theme = {
+    persistentMenu: [
+      { label: 'Help', payload: 'help' },
+      {
+        label: 'See docs',
+        url: 'https://docs.botonic.io',
+      },
+      { closeLabel: 'Close' },
+    ],
+    enableEmojiPicker: true,
+    enableAttachments: true,
+  }
+
+  window.HTMLElement.prototype.scrollTo = function () {}
+
+  it('Webchat has by default', async () => {
+    const { result } = renderUseWebchatHook()
+    await act(async () => {
+      render(<Webchat webchatHooks={result.current} />)
+    })
+    expect(screen.queryByRole('trigger-button')).toBeTruthy()
+    expect(screen.queryByRole('styled-webchat')).toBeNull()
+  })
+
+  it('Opened webchat has by default', async () => {
+    const { result } = renderUseWebchatHook()
+    act(() => {
+      result.current.toggleWebchat(true)
+    })
+    await act(async () => {
+      render(<Webchat webchatHooks={result.current} />)
+    })
+    expect(screen.queryByRole('styled-webchat')).toBeTruthy()
+    expect(screen.queryByRole('header')).toBeTruthy()
+    expect(screen.queryByRole('message-list')).toBeTruthy()
+    expect(screen.queryByRole('textbox')).toBeTruthy()
+    expect(screen.queryByRole('send-button-icon')).toBeTruthy()
+  })
+
+  it('Opened webchat has not by default', async () => {
+    const { result } = renderUseWebchatHook()
+    act(() => {
+      result.current.toggleWebchat(true)
+    })
+    await act(async () => {
+      render(<Webchat webchatHooks={result.current} />)
+    })
+    expect(screen.queryByRole('trigger-button')).toBeNull()
+    expect(screen.queryByRole('persistent-menu-icon')).toBeNull()
+    expect(screen.queryByRole('emoji-picker-icon')).toBeNull()
+    expect(screen.queryByRole('attachment-icon')).toBeNull()
+  })
+
+  it('UpdateTyping', async () => {
+    const { result } = renderUseWebchatHook()
+    act(() => {
+      result.current.toggleWebchat(true)
+      result.current.updateTyping(true)
+    })
+    await act(async () => {
+      render(<Webchat webchatHooks={result.current} />)
+    })
+    expect(screen.queryByRole('typing-indicator')).toBeTruthy()
+  })
+
+  it('Enable emojiPicker, attachments and define persistent menu', async () => {
+    const { result } = renderUseWebchatHook()
+    Object.assign(theme, result.current.webchatState.theme)
+    act(() => {
+      result.current.toggleWebchat(true)
+      result.current.updateTheme(theme)
+    })
+    await act(async () => {
+      render(<Webchat webchatHooks={result.current} />)
+    })
+    expect(screen.queryByRole('textbox')).toBeTruthy()
+    expect(screen.queryByRole('persistent-menu-icon')).toBeTruthy()
+    expect(screen.queryByRole('emoji-picker-icon')).toBeTruthy()
+    expect(screen.queryByRole('attachment-icon')).toBeTruthy()
+    expect(screen.queryByRole('send-button-icon')).toBeTruthy()
+  })
+
+  it.skip('Open EmojiPicker', async () => {
```

Comment:
for newcoming pr reviewers, we have decided to skip these tests atm and we will keep continue working on that on new tasks.